aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -98,6 +98,25 @@ Tensor expand_batching_rule(const Tensor& self, IntArrayRef size, bool implicit)
   return self_physical.newLogicalFromPhysical(result);
 }
 
+Tensor unsqueeze_batching_rule(const Tensor& self, int64_t dim) {
+  auto self_physical = MultiBatchVmapTransform::logicalToPhysical(self);
+  // NB: unsqueeze has some special handling of its `dim` argument so we can't call
+  // self_physical.getPhysicalDim. Semantically speaking `dim` is the dimension
+  // index on the NEW tensor where the new dimension appears.
+  auto dim_physical =
+      self_physical.numBatchDims() + maybe_wrap_dim(dim, /*logical_dim*/self.dim() + 1);
```

Comment:
Why is this "self.dim() + 1"?

Reply:
Unsqueeze wraps to one plus the logical dim: https://github.com/pytorch/pytorch/blob/b623bdeabb0aa8da44285d303246e7f8ac06c2a9/aten/src/ATen/native/TensorShape.cpp#L1413

I'll make that clearer in the comment and link to said code

Reply:
comment added